openslides/assignment/models.py

```diff
@@ -80,125 +144,115 @@ def get_absolute_url(self, link='detail'):
-            return False
+        return self.related_users.filter(
+            assignmentrelateduser__status=AssignmentRelatedUser.STATUS_BLOCKED)
 
-    def is_blocked(self, person):
+    def is_candidate(self, user):
         """
-        return True, if the person is blockt for candidature.
+        Returns True if user is a candidate.
+
+        Costs one database query.
         """
-        return self.assignment_candidates.filter(person=person).filter(blocked=True).exists()
+        return self.candidates.filter(pk=user.pk).exists()
 
-    @property
-    def assignment_candidates(self):
-        return AssignmentCandidate.objects.filter(assignment=self)
+    def is_elected(self, user):
+        """
+        Returns True if the user is elected for this assignment.
 
-    @property
-    def candidates(self):
-        return self.get_participants(only_candidate=True)
+        Costs one database query.
+        """
+        return self.elected.filter(pk=user.pk).exists()
 
-    @property
-    def elected(self):
-        return self.get_participants(only_elected=True)
+    def is_blocked(self, user):
+        """
+        Returns True if the user is blockt for candidature.
+
+        Costs one database query.
+        """
+        return self.blocked.filter(pk=user.pk).exists()
 
-    def get_participants(self, only_elected=False, only_candidate=False):
-        candidates = self.assignment_candidates.exclude(blocked=True)
+    def set_candidate(self, user, force=False):
+        """
+        Adds the user as candidate.
+        """
+        related_user, __ = self.assignment_related_users.update_or_create(
+            user=user,
+            defaults={'status': AssignmentRelatedUser.STATUS_CANDIDATE})
 
-        assert not (only_elected and only_candidate)
+    def set_elected(self, user):
+        """
+        Makes user an elected user for this assignment.
+        """
+        related_user, __ = self.assignment_related_users.update_or_create(
+            user=user,
+            defaults={'status': AssignmentRelatedUser.STATUS_ELECTED})
 
-        if only_elected:
-            candidates = candidates.filter(elected=True)
+    def set_blocked(self, user):
+        """
+        Block user from this assignment, so he can not get an candidate.
+        """
+        related_user, __ = self.assignment_related_users.update_or_create(
+            user=user,
+            defaults={'status': AssignmentRelatedUser.STATUS_BLOCKED})
 
-        if only_candidate:
-            candidates = candidates.filter(elected=False)
+    def delete_related_user(self, user):
+        """
+        Delete the connection from the assignment to the user.
+        """
+        self.assignment_related_users.filter(user=user).delete()
 
-        # TODO: rewrite this with a queryset
-        participants = []
-        for candidate in candidates.all():
-            participants.append(candidate.person)
-        return participants
+    def set_phase(self, phase):
```

Comment:
No. We decided that models should not save itself but that the code that calls the method has to call model.save() explicit .

In other case
```python
model.set_foo()
model.set_bar()
```
would save model two times.